packages/docs/src/routes/docs/deployments/self-hosting/index.mdx

```diff
@@ -0,0 +1,536 @@
+FROM deps as build
+
+# Copy the rest of the source files into the image.
+COPY . .
+
+# Run the build script.
+RUN yarn run build
+
+################################################################################
+# Create a new stage to run the application with minimal runtime dependencies
+FROM base as final
+
+# Use production node environment by default.
+ENV NODE_ENV production
+
+# IMPORTANT: Set your actual domain for CSRF protection
+ENV ORIGIN https://your-domain.com
+
+# Run the application as a non-root user.
+USER node
+
+# Copy package.json so that package manager commands can be used.
+COPY package.json .
+
+# Copy the production dependencies from the deps stage and also
+# the built application from the build stage into the image.
+COPY --from=deps /usr/src/app/node_modules ./node_modules
+COPY --from=build /usr/src/app/dist ./dist
+COPY --from=build /usr/src/app/server ./server
+
+# Expose the port that the application listens on.
+EXPOSE 3000
+
+# Run the application.
+CMD yarn serve
+```
+
+Build and run your Docker container:
+
+```shell
+# Build the image
+docker build -t qwik-app .
+
+# Run the container
+docker run -p 3000:3000 -e ORIGIN=https://your-domain.com qwik-app
+```
+
+### Method 3: Static Site Deployment
+
+If using the static adapter, copy the contents of the `dist` directory to your web server's document root:
+
+```shell
+# Example using rsync
+rsync -avz dist/ user@your-server:/path/to/webroot/
+```
+
+## Web Server Configuration
+
+### Nginx Configuration for Node.js Adapters
+
+For Node.js deployments using Express or Fastify, use Nginx as a reverse proxy:
+
+```nginx
+server {
+    listen 80;
+    server_name your-domain.com;
+    
+    # Redirect HTTP to HTTPS
+    location / {
+        return 301 https://$host$request_uri;
+    }
+}
+
+server {
+    listen 443 ssl http2;
+    server_name your-domain.com;
+
+    # SSL configuration
+    ssl_certificate /path/to/cert.pem;
+    ssl_certificate_key /path/to/key.pem;
+    
+    # SSL settings (recommended)
+    ssl_protocols TLSv1.2 TLSv1.3;
+    ssl_prefer_server_ciphers on;
+    ssl_ciphers ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-CHACHA20-POLY1305:ECDHE-RSA-CHACHA20-POLY1305:DHE-RSA-AES128-GCM-SHA256:DHE-RSA-AES256-GCM-SHA384;
+    ssl_session_timeout 1d;
+    ssl_session_cache shared:SSL:10m;
+
+    # Proxy to Node.js server
+    location / {
+        proxy_pass http://localhost:3000;
+        proxy_http_version 1.1;
+        proxy_set_header Upgrade $http_upgrade;
+        proxy_set_header Connection 'upgrade';
+        proxy_set_header Host $host;
+        proxy_cache_bypass $http_upgrade;
+        proxy_set_header X-Real-IP $remote_addr;
+        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
+        proxy_set_header X-Forwarded-Proto $scheme;
+    }
+
+    # Serve static assets directly for better performance
+    location /build/ {
```

Comment:
also add `/assets/` and explain that `/some/app/route/q-data.json` files contain routeLoader data and should also get appropriate caching headers, e.g. 30 seconds for data that must be fresh.